x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/FieldSubsetReader.java

```diff
@@ -399,6 +400,20 @@ public void binaryField(FieldInfo fieldInfo, byte[] value) throws IOException {
                 Map<String, Object> transformedSource = filter(result.v2(), filter, 0);
                 XContentBuilder xContentBuilder = XContentBuilder.builder(result.v1().xContent()).map(transformedSource);
                 visitor.binaryField(fieldInfo, BytesReference.toBytes(BytesReference.bytes(xContentBuilder)));
+            } else if (IgnoredSourceFieldMapper.NAME.equals(fieldInfo.name)) {
```

Comment:
Maybe add a unit test for this logic to `FieldSubsetReaderTests`? Things like _source filtering are also unit tested there.

Reply:
Source is easy to read and write, but  `_ignored_source` contents are encoded..

I think this requires either exposing parts of `IgnoredSourceFieldMapper` and `XContentDataHelper` as public, or making this a `MapperServiceTestCase` subclass. Any preference?

Reply:
I think making some methods of `IgnoredSourceFieldMapper` and `XContentDataHelper ` public for the  purpose of testing is ok.

Reply:
I went with `MapperServiceTestCase` to check more code paths, still not short..